asab/web/authn/oauth.py

```diff
@@ -4,25 +4,24 @@
 import asab.web.rest
 
 
-async def oauthclient_middleware_factory(request, handler, url):
-	authorization = request.headers.get("Authorization", None)
-	if authorization is None:
-		return asab.web.rest.json_response(request, {
-			"result": "AUTHORIZATION-NOT-PROVIDED"
-		})
-
-	async with aiohttp.ClientSession() as session:
-		async with session.get(url, headers={"Authorization": authorization}) as resp:
-			if resp.status == 200:
-				identity = await resp.json()
-				if identity is not None:
-					request.Identity = identity
-					return await handler(request)
-			else:
-				return asab.web.rest.json_response(request, {
-					"result": "AUTHORIZATION-FAILED"
-				})
-
-	return asab.web.rest.json_response(request, {
-		"result": "AUTHORIZATION-ERROR"
-	})
\ No newline at end of file
+def oauthclient_middleware_factory(app, *args, url, **kwargs):
+
+	@aiohttp.web.middleware
+	async def oauthclient_middleware(request, handler):
+
+		authorization = request.headers.get(aiohttp.hdrs.AUTHORIZATION, None)
+		if authorization is None:
+			return await handler(request)
+
+		async with aiohttp.ClientSession() as session:
+			async with session.get(url, headers={"Authorization": authorization}) as resp:
+				if resp.status == 200:
+					identity = await resp.json()
+					if identity is not None:
+						request.Identity = identity
```

Comment:
Identity should be a string (unique).

Reply:
You may store the token (or what is a proper name) in `request.OAuthToken = ` etc.

Reply:
Well, in our case yes, but generally the user info endpoint is expected to return a JSON with user identification such as login etc. https://connect2id.com/products/server/docs/api/userinfo

Reply:
You are mixing OAuth with OpenID Connect ... JSON is expected in OpenID Connect.
`request.OAuthUserInfo = ` will be ok, then.